filters/parser.go

```diff
@@ -42,7 +42,7 @@ The formal grammar is as follows:
 selectors := selector ("," selector)*
 selector  := fieldpath (operator value)
 fieldpath := field ('.' field)*
-field     := quoted | [A-Za-z] [A-Za-z0-9_]+
+field     := quoted | [A-Za-z-] [A-Za-z0-9_]+
```

Comment:
Is `-` allowed only for the first letter of a field?

Reply:
This logic is incorrect and `-` should only be allowed as a separator similar to `.`